pkg/asset/agent/installconfig.go

```diff
@@ -120,3 +124,172 @@ func (a *OptionalInstallConfig) ClusterName() string {
+	}
+
+	for i, compute := range installConfig.Compute {
+		if compute.Hyperthreading != "Enabled" {
+			fieldPath := field.NewPath(fmt.Sprintf("Compute[%d]", i), "Hyperthreading")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, compute.Hyperthreading))
+		}
+		// kubebuilder:default=amd64. Set from generic install config code
+		if compute.Architecture != types.ArchitectureAMD64 && compute.Architecture != "x86_64" {
+			fieldPath := field.NewPath(fmt.Sprintf("Compute[%d]", i), "Architecture")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, compute.Architecture))
+		}
+
+		if compute.Platform.AWS != nil || compute.Platform.AlibabaCloud != nil || compute.Platform.Azure != nil ||
+			compute.Platform.BareMetal != nil || compute.Platform.GCP != nil || compute.Platform.IBMCloud != nil ||
+			compute.Platform.Libvirt != nil || compute.Platform.Nutanix != nil || compute.Platform.OpenStack != nil ||
+			compute.Platform.Ovirt != nil || compute.Platform.PowerVS != nil || compute.Platform.VSphere != nil {
+			fieldPath := field.NewPath(fmt.Sprintf("Compute[%d]", i), "Platform")
+			logrus.Warnf(fmt.Sprintf("%s is ignored", fieldPath))
+		}
+	}
+
+	if installConfig.ControlPlane.Hyperthreading != "Enabled" {
+		fieldPath := field.NewPath("ControlPlane", "Hyperthreading")
+		logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, installConfig.ControlPlane.Hyperthreading))
+	}
+
+	// kubebuilder:default=amd64. Set from generic install config code
+	if installConfig.ControlPlane.Architecture != types.ArchitectureAMD64 && installConfig.ControlPlane.Architecture != "x86_64" {
+		fieldPath := field.NewPath("ControlPlane", "Architecture")
+		logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, installConfig.ControlPlane.Architecture))
+	}
+
+	if installConfig.ControlPlane.Platform.AWS != nil || installConfig.ControlPlane.Platform.AlibabaCloud != nil ||
+		installConfig.ControlPlane.Platform.Azure != nil || installConfig.ControlPlane.Platform.BareMetal != nil ||
+		installConfig.ControlPlane.Platform.GCP != nil || installConfig.ControlPlane.Platform.IBMCloud != nil ||
+		installConfig.ControlPlane.Platform.Libvirt != nil || installConfig.ControlPlane.Platform.Nutanix != nil ||
+		installConfig.ControlPlane.Platform.OpenStack != nil || installConfig.ControlPlane.Platform.Ovirt != nil ||
+		installConfig.ControlPlane.Platform.PowerVS != nil || installConfig.ControlPlane.Platform.VSphere != nil {
+		fieldPath := field.NewPath("ControlPlane", "Platform")
+		logrus.Warnf(fmt.Sprintf("%s is ignored", fieldPath))
+	}
+
+	switch installConfig.Platform.Name() {
+	case baremetal.Name:
+		baremetal := installConfig.Platform.BareMetal
+		// +kubebuilder:default="qemu:///system". Set from generic install config code
+		if baremetal.LibvirtURI != "qemu:///system" {
+			fieldPath := field.NewPath("Platform", "Baremetal", "LibvirtURI")
+			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.LibvirtURI))
```

Comment:
This is only used for the bootstrap VM, which people (hopefully) know we don't need. So we could probably safely reduce this to debug level.